### PR TITLE
made a new SingleArticlePage file to render when one of the articles …

### PR DESCRIPTION
**Pull request by Antonio-Bogalho**
…on HomePage is clicked, added routes on Main Routes, made a new function to fetch an article by its id on my utils file, next step is to work on displaying the comments on a SingleArticlePage

### CODE REVIEW

File: components/ArticleCard.jsx [hunk 1]
```diff
@@ -1,8 +1,8 @@
-
-
+import { Link } from "react-router-dom"
 export default function ArticleCard({ article }){
     return (
         <li>
+            <Link to={`/articles/${article.article_id}`} >
```
Comment (nc-mark-t) [inline components/ArticleCard.jsx:5]: I would recommend using prettier to format your code before a PR

File: components/HomePage.jsx [hunk 2]
```diff
@@ -5,15 +5,29 @@ import ArticleCard from "./ArticleCard"
...
     const [articlesList, setArticlesList] = useState([])
+    const [isLoading, setIsLoading] = useState(false)
+    const [isError, setIsError] = useState(false)
 
     useEffect(() => {
+        setIsLoading(true)
+        setIsError(false)
         fetchAllArticles()
         .then((response) => {
             console.log(response)
```
Comment (nc-mark-t) [inline components/HomePage.jsx:16]: be sure to remove console.logs

File: components/SingleArticlePage.jsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            <img src={singleArticle.article_img_url}/>
+            <p>Votes: {singleArticle.votes}</p>
+            <p>Comments: {singleArticle.comment_count}</p>
+            <button className="item-card-buy-btn">+1 vote</button>
+            <button className="item-card-buy-btn">-1 vote</button>
+            <p>Created at: {singleArticle.created_at}</p>
+            <p>Test text to see if single card is unique, add comments to it later</p>
+
+        </section>
+        </>
```
Comment (nc-mark-t) [inline components/SingleArticlePage.jsx:48]: This fragment in unnecessary as the section is a parent of all the other elements

File: components/SingleArticlePage.jsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import { fetchArticleById } from "../utils/api"
+
+export default function SingleArticlePage(){
+    const [singleArticle, setSingleArticle] = useState({})
+    const [isLoading, setIsLoading] = useState(false)
+    const [isError, setIsError] = useState(false)
+    const { article_id} = useParams()
+
+    useEffect(() => {
+        if (article_id){
```
Comment (nc-mark-t) [inline components/SingleArticlePage.jsx:12]: You should always have an article id because without it the router will never match this route

**Review by nc-mark-t — COMMENTED** [review]
Good work

**Comment by Antonio-Bogalho** [comment]
Thank you for your feedback, I'll make sure to have it done by the next
ticket (7) :)

**Pull request merged**